lib/gollum-lib/filter/tags.rb

```diff
@@ -189,6 +188,7 @@ def process_external_link_tag(url, pretty_name = nil)
   # Returns the String HTML if the tag is a valid file link tag or nil
   #   if it is not.
   def process_file_link_tag(link_part, pretty_name)
+    return nil if ::Gollum::Page.valid_extension?(link_part)
```

Comment:
Managed to catch a nasty edge case here when I added one of the tests for link names. If you'd specify a link to a page including its file extension, the result would be a **file** link, not an **absent page** link, because `process_file_link_tag` would succeed (since all Pages are also Files).

Reply:
This is when the page in question does not exist, correct?

Reply:
It's a little more complicated unfortunately. At this point in the chain, we have not yet tested whether the page exists. We need to test for a matching **File** before we test for a matching **Page**, because otherwise we could never get the result that the link is an "internal absent" link. If there's not a Page, there could yet be a file, so we can't mark it as absent on that. But File links aren't marked as absent or present at all.

So what this does is: it stops the tag filter from treating something as a link to a File if it has a valid Page extension. Then we go on to evaluate the link as a possible Page extension, and we check for Page existence.

An alternative to this could be to completely eradicate the difference between Page and File links: just look for a File in all cases, and mark the link as "internal present" or "internal absent".

Reply:
> An alternative to this could be to completely eradicate the difference between Page and File links: just look for a File in all cases, and mark the link as "internal present" or "internal absent".

Just realized this is not possible, as the file-finding algorithm doesn't have the required "slack" for matching page extensions to actually find any pages. :(

Hope this fix seems ok!